### PR TITLE
Spread props in executeRootBundle to improve etherscan compatibility

### PR DESCRIPTION
**Pull request by chrismaree**

### CODE REVIEW

File: contracts/HubPool.sol [hunk 1]
```diff
@@ -549,64 +549,70 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
...
+     * @param chainId ChainId number of the target spoke pool on which the bundle is executed.
+     * @param bundleLpFees Array representing the total LP fee amount per token in this bundle for all bundled relays.
+     * @param netSendAmounts Array representing the amount of tokens to send to the SpokePool on the target chainId.
+     * @param runningBalances Array used to track any unsent tokens that are not included in the netSendAmounts.
+     * @param leafId Index of this executed leaf within the poolRebalance tree.
+     * @param l1Tokens Array of all the tokens associated with the bundleLpFees, nedSendAmounts and runningBalances.
      * @param proof Inclusion proof for this leaf in pool rebalance root in root bundle.
      */
-    function executeRootBundle(PoolRebalanceLeaf memory poolRebalanceLeaf, bytes32[] memory proof) public nonReentrant {
+    function executeRootBundle(
```
Comment (chrismaree) [inline contracts/HubPool.sol:560]: This is the primary change of this PR: spread the `PoolRebalanceLeaf` object to improve etherscan tx processing.

File: contracts/HubPool.sol [hunk 2]
```diff
@@ -549,64 +549,70 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
...
         require(getCurrentTime() > rootBundleProposal.requestExpirationTimestamp, "Not passed liveness");
 
         // Verify the leafId in the poolRebalanceLeaf has not yet been claimed.
-        require(!MerkleLib.isClaimed1D(rootBundleProposal.claimedBitMap, poolRebalanceLeaf.leafId), "Already claimed");
+        require(!MerkleLib.isClaimed1D(rootBundleProposal.claimedBitMap, leafId), "Already claimed");
 
         // Verify the props provided generate a leaf that, along with the proof, are included in the merkle root.
         require(
-            MerkleLib.verifyPoolRebalance(rootBundleProposal.poolRebalanceRoot, poolRebalanceLeaf, proof),
+            MerkleLib.verifyPoolRebalance(
```
Comment (chrismaree) [inline contracts/HubPool.sol:576]: loose some lines here.

File: contracts/HubPool.sol [hunk 3]
```diff
@@ -549,64 +549,70 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
...
-        rootBundleProposal.claimedBitMap = MerkleLib.setClaimed1D(
-            rootBundleProposal.claimedBitMap,
-            poolRebalanceLeaf.leafId
-        );
+        rootBundleProposal.claimedBitMap = MerkleLib.setClaimed1D(rootBundleProposal.claimedBitMap, leafId);
 
         // Decrement the unclaimedPoolRebalanceLeafCount.
         rootBundleProposal.unclaimedPoolRebalanceLeafCount--;
 
-        _sendTokensToChainAndUpdatePooledTokenTrackers(
```
Comment (chrismaree) [inline contracts/HubPool.sol:587]: win some lines here.

**Review by nicholaspai — APPROVED** [review]
Does this affect gas at all?

File: contracts/HubPool.sol [hunk 4]
```diff
@@ -549,64 +549,70 @@ contract HubPool is HubPoolInterface, Testable, Lockable, MultiCaller, Ownable {
...
         // Verify the leafId in the poolRebalanceLeaf has not yet been claimed.
-        require(!MerkleLib.isClaimed1D(rootBundleProposal.claimedBitMap, poolRebalanceLeaf.leafId), "Already claimed");
+        require(!MerkleLib.isClaimed1D(rootBundleProposal.claimedBitMap, leafId), "Already claimed");
 
         // Verify the props provided generate a leaf that, along with the proof, are included in the merkle root.
         require(
-            MerkleLib.verifyPoolRebalance(rootBundleProposal.poolRebalanceRoot, poolRebalanceLeaf, proof),
+            MerkleLib.verifyPoolRebalance(
+                rootBundleProposal.poolRebalanceRoot,
+                PoolRebalanceLeaf({
```
Comment (mrice32) [inline contracts/HubPool.sol:578]: Just OOC, how does the gas of this method compare (I think we have gas tests for execution now)? I worry that with large arrays, we may end up incurring significant gas costs when copying. Also not sure if structs can contain references to memory or have to copy all their constituents.

**Pull request merged**